joomla/joomla-cms
# Fallback for keepalive js

Pull Request for Issue https://github.com/joomla/joomla-cms/issues/16375 (only keepalive fallback part).

### Summary of Changes

The keepalive js fallback is not working correctly when there is no `system.keepalive` json. This PR intends to add better fallback for those wild scenarios.

### Testing Instructions

Pre-requisites:
- Use latest joomla staging with a login module in the frontend (so keepalive is called)
- In global config set joomla session to 1 minute (for faster test) and System Cache to "ON - Progressive caching"

Reproducing the issue:
- Clear joomla cache
- Go to frontend - not logged, reload the page two times (Ctrl+F5)
- Confirm there is a **js error** in the Chrome Debug Console "Console" tab (F12)

Test if proposed keepalive fallback works:
- Apply patch https://github.com/joomla/joomla-cms/pull/16488 (if not merged) and them this patch
- Repeat steps in "Reproducing the issue", but now confirm there is **no js error** in the Chrome Debug Console "Console" tab (F12)
- Wait one minute and also confirm keepalive in called correctly in Chrome Debug Console "Network" tab

### Expected result

Keepalive should always work even if there is no `system.keepalive` json.

### Actual result

Keepalive does not work in this scenario because of a js error.

### Documentation Changes Required

@dgt41 @Fedik can you also check this one and confirm there is no problem in adding the joomla install uri paths trough script options in behavior.core, ie, all the pages that use the core bahaviour (this would also be used in https://github.com/joomla/joomla-cms/pull/15529).
